packages/runtime/src/runtime.ts

```diff
@@ -471,80 +500,60 @@ export class Runtime {
    * @param args : external arguments to smart contract
    */
-  /* eslint-disable sonarjs/cognitive-complexity */
-  executeTx (txnParams: ExecParams | ExecParams[], program: string,
-    args: Uint8Array[]): void {
+  executeTx (txnParams: ExecParams | ExecParams[], args?: Uint8Array[]): void {
     const [tx, gtxs] = this.createTxnContext(txnParams); // get current txn and txn group (as encoded obj)
 
     // initialize context before each execution
     this.ctx = {
       state: cloneDeep(this.store), // state is a deep copy of store
       tx: tx,
       gtxs: gtxs,
-      args: args
+      args: args ?? []
     };
 
-    // TODO: Add Support for group transaction for execution modes
-    // as for stateless TEAL we need to have TEAL code with each txParam
-    // TASKs: https://www.pivotaltracker.com/story/show/176455371, https://www.pivotaltracker.com/story/show/176455329
-    let mode = ExecutionMode.STATEFUL;
-    if (!Array.isArray(txnParams)) {
-      mode = this.getExecutionMode(txnParams);
-      this.assertAmbiguousTxnParams(txnParams);
-      if (txnParams.sign === SignType.LogicSignature) {
-        this.validateLsigAndRun(txnParams);
-      }
-    } else {
-      let flag = true;
-      for (const txParam of txnParams) {
-        this.assertAmbiguousTxnParams(txParam);
-        if (txParam.sign === SignType.LogicSignature) {
-          this.validateLsigAndRun(txParam);
-        }
-        flag = flag && txParam.sign === SignType.LogicSignature;
-      }
-      if (flag) { mode = ExecutionMode.STATELESS; } // if all txns in grp are stateless
-    }
-
     const txnParameters = Array.isArray(txnParams) ? txnParams : [txnParams];
-    try {
-      this.run(program, mode);
-    } catch (error) {
-      // if transaction type is Clear Call, remove the app first before throwing error (rejecting tx)
+    // Run TEAL program associated with each transaction without interacting with store.
+    for (const txnParam of txnParameters) {
+      this.assertAmbiguousTxnParams(txnParam);
+      if (txnParam.sign === SignType.LogicSignature) {
+        this.validateLsigAndRun(txnParam);
+      }
+      let fromAccount;
       // https://developer.algorand.org/docs/features/asc1/stateful/#the-lifecycle-of-a-stateful-smart-contract
-      for (const txnParam of txnParameters) {
-        if (txnParam.type === TransactionType.ClearSSC) {
-          const fromAccount = this.getAccount(txnParam.fromAccount.addr);
+      switch (txnParam.type) {
+        case TransactionType.CallNoOpSSC:
+          this.run(this.getProgram(txnParam.appId)[0], ExecutionMode.STATEFUL); // approval program
+          break;
+        case TransactionType.CloseSSC:
+          this.run(this.getProgram(txnParam.appId)[0], ExecutionMode.STATEFUL); // approval program
+          break;
+        case TransactionType.DeleteSSC:
+          this.run(this.getProgram(txnParam.appId)[0], ExecutionMode.STATEFUL); // approval program
+          break;
```

Comment:
Instead of repeating 3 times can we do:
```
 case TransactionType.CallNoOpSSC || CloseSSC || DeleteSSC:
   this.run(this.getProgram(txnParam.appId)[0], ExecutionMode.STATEFUL); // approval program
   break;
```

Reply:
yes